packages-web/datagrid/src/components/Pagination.tsx

```diff
@@ -0,0 +1,65 @@
+import { createElement, ReactElement, Dispatch, SetStateAction } from "react";
+
+export interface PaginationProps {
+    page: number;
+    setPage: Dispatch<SetStateAction<number>>;
+    hasMoreItems: boolean;
+}
+
+export interface ClientSidePaginationProps {
+    gotoPage: (page: number) => void;
+    previousPage: () => void;
+    nextPage: () => void;
+    page: number;
+    pageOptions: number[];
+    canPreviousPage: boolean;
+    canNextPage: boolean;
+}
+
+export function Pagination({ page, setPage, hasMoreItems }: PaginationProps): ReactElement {
```

Comment:
Note that for server-side paging you also might know the total number of items => total number of pages => you can have exactly the same UI for client and server paging, if you gracefully handle the fact that you don't know the total number of pages.

Reply:
Merged 2 in one :)